library/math/count_paths/count_paths_triangle.hpp

```diff
@@ -0,0 +1,30 @@
+/** @file */
+#pragma once
+#include "count_paths_rectangle.hpp"
+/**
+ * @see https://noshi91.hatenablog.com/entry/2023/07/21/235339
+ * @param h,bottom see image
+ * @returns see image
+ * @time O((n + m)log^2(n + m)): n = sz(h); m = h.back()
+ * @space O(n + m)
+ */
+vl divide_and_conquer(vl h, vl bottom) {
+  assert(sz(h) == sz(bottom) && is_sorted(all(h)) && !empty(h) && h[0] >= 0);
+  {
+    int start = find_if(all(h), [](ll x) { return x; }) - begin(h);
+    h.erase(begin(h), begin(h) + start);
+    bottom.erase(begin(bottom), begin(bottom) + start);
+  }
+  int n = sz(h);
+  if (n == 0) return {};
+  if (n == 1) return vl(h[0], bottom[0]);
+  int mid = n / 2;
+  auto left = divide_and_conquer(vl(begin(h), begin(h) + mid),
+                                 vl(begin(bottom), begin(bottom) + mid));
+  auto [right, top] = get_right_and_top(left, vl(mid + all(bottom)));
+  vl h_right(mid + all(h));
+  for (ll& hh : h_right) hh -= h[mid - 1];
+  auto upper_right = divide_and_conquer(h_right, top);
+  right.insert(end(right), all(upper_right));
```

Comment:
do you like these var names? we were writing them kinda quick so if you think they could be better definitely change them